### ISSUE TITLE
livenessProbe should log useful failure messages

### ISSUE THREAD

**Issue by davidhyman** [kubernetes/kubernetes#46218]
FEATURE REQUEST
**Kubernetes version**:
GitVersion:"v1.6.2"

**Environment**:
Google Container Engine

**What happened**:
Various issues getting the url / timeout etc params correct for the deployment (also see https://github.com/kubernetes/kubernetes/issues/29470 or https://github.com/kubernetes/kubernetes/issues/29470)
Very long debug cycle.
Right now the only messages one can get out of the system (e.g. with `kubectl describe` or using `kubectl proxy` are along the lines of:
```
Killing container with id docker://70a8xxx:pod "my-pod-3965xxx_my-namespace(9b28710e-xxx)" container "my-container" is unhealthy, it will be killed and re-created.
```
for the httpGet endpoint, errors might be (as a small example - there are probably more!):
- 404/500 / other non-200 OK response from server
- probe request timeout (server took too long)
- wrong schema, so can't make connection
- wrong host, so can't make connection
- wrong port, so can't make connection

_but which one was it?_

Using `kubectl exec -ti ...` to get in to the pod and try the liveness probe directly does not necessarily help with debugging because it does not take in to account how Kubernetes interprets the Deployment configuration (i.e. intention != outcome).

**What you expected to happen**:
Kubernetes should log more detailed messages when choosing to kill a pod (which is a fairly drastic action to be taking).
As an example, a minimum, sufficient solution could be an 'error' object included in the logs. 
`container "my-container" is unhealthy (livenessProbe.httpGet: {status: 404})`
`container "my-container" is unhealthy (livenessProbe.httpGet: {timeout: 3.348})`
Presumably a new/different data container or log message is additionally needed to allow programmatic use.

**How to reproduce it** (as minimally and precisely as possible):
Apply a httpGet liveness probe to a pod with various failure modes (404, 500, request timeout etc).
https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#defining-a-liveness-http-request

**Anything else we need to know**:
Raised against Kubernetes - if this behaviour is already implemented and just needs exposing that's great! In which case this ticket can move to kubectl.

**Comment by tacy** [comment]
@apelisse - Could you relable it as sig/node, it releated to kubelet

@davidhyman - Let me take a look, if so, I will fix it

**Comment by tacy** [comment]
@davidhyman  -  it has been fixed by #46246.
